sklearn/preprocessing/data.py

```diff
@@ -250,20 +312,74 @@ def fit(self, X, y=None):
         X : array-like, shape [n_samples, n_features]
             The data used to compute the per-feature minimum and maximum
             used for later scaling along the features axis.
+            X cannot be a sparse matrix.
         """
+        if sparse.issparse(X):
+            raise TypeError("MinMaxScaler does no support sparse input. "
+                            "You may consider to use MaxAbsScaler instead.")
+
         X = check_array(X, copy=self.copy, ensure_2d=False, warn_on_dtype=True,
                         estimator=self, dtype=FLOAT_DTYPES)
         feature_range = self.feature_range
         if feature_range[0] >= feature_range[1]:
             raise ValueError("Minimum of desired feature range must be smaller"
                              " than maximum. Got %s." % str(feature_range))
+
         data_min = np.min(X, axis=0)
-        data_range = np.max(X, axis=0) - data_min
-        data_range = _handle_zeros_in_scale(data_range)
-        self.scale_ = (feature_range[1] - feature_range[0]) / data_range
+        data_max = np.max(X, axis=0)
+        data_range = data_max - data_min
+        self.scale_ = ((feature_range[1] - feature_range[0]) /
+                       _handle_zeros_in_scale(data_range))
         self.min_ = feature_range[0] - data_min * self.scale_
-        self.data_range = data_range
```

Comment:
For some reasons those attributes do not appear in the docstring. Do we deprecate them anyway?